src/schedule.rs

```diff
@@ -372,11 +385,22 @@ where
     fn next(&mut self) -> Option<DateTime<Z>> {
         let previous = self.previous_datetime.take()?;
 
-        if let Some(next) = self.schedule.next_after(&previous) {
-            self.previous_datetime = Some(next.clone());
-            Some(next)
+        if let Some(later) = self.later_datetime.take() {
+            self.previous_datetime = Some(later.clone());
+            Some(later)
         } else {
-            None
+            match self.schedule.next_after(&previous) {
+                LocalResult::Single(next) => {
+                    self.previous_datetime = Some(next.clone());
+                    Some(next)
+                }
+                LocalResult::Ambiguous(earlier, later) => {
+                    self.previous_datetime = Some(earlier.clone());
+                    self.later_datetime = Some(later);
+                    Some(earlier)
+                }
+                LocalResult::None => None,
```

Comment:
This keeps mostly the same functionality as the old `if let` statement, but also handles the ambiguous case, to store the next one.